src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfileDataFlowExample.cs

```diff
@@ -0,0 +1,134 @@
+    ///      user know that things may not work properly.
+    ///   3. Importing an UnconfiguredProject in the constructor also guarantees we get a new instance of this type
+    ///      associated with each UnconfiguredProject (though this is a bit redundant as the other two imports would
+    ///      also cause this).
+    /// </remarks>
+    private readonly UnconfiguredProject _project;
+    /// <remarks>
+    /// We need the ILaunchSettingsProvider in order to hook up to ILaunchSettingsProvider.SourceBlock.
+    /// </remarks>
+    private readonly ILaunchSettingsProvider _launchSettingsProvider;
+    /// <remarks>
+    /// The IProjectPropertiesProvider is needed to update the project properties when the launch settings change.
+    /// </remarks>
+    private readonly IProjectPropertiesProvider _projectProperties;
+
+    /// <remarks>
+    /// Hooking up to a data flow source creates a link; disposing the link is how to break the connection to the data
+    /// flow source when we no longer need it.
+    /// </remarks>
+    private IDisposable? _launchSettingsLink;
+    /// <remarks>
+    /// We need to store some state, namely, the previous command name/debug target.
+    /// </remarks>
+    private string? _previousCommandName = null;
+
+    [ImportingConstructor]
+    public LaunchProfileDataFlowExample(
+        UnconfiguredProject unconfiguredProject,
+        ILaunchSettingsProvider launchSettingsProvider,
+        // There are multiple implementations of IProjectPropertiesProvider. The Import attribute here tells MEF to give
+        // us the specific one named "ProjectFile".
+        [Import(ContractNames.ProjectPropertyProviders.ProjectFile)] IProjectPropertiesProvider projectProperties)
+    {
+        _project = unconfiguredProject;
+        _launchSettingsProvider = launchSettingsProvider;
+        _projectProperties = projectProperties;
+    }
+
+    /// <remarks>
+    /// IProjectDynamicLoadComponent.LoadAsync is called every time the project satisifies the capabilities specified via
+    /// AppliesTo. This is where we hook up to the data flow source.
+    /// </remarks>
+    public Task LoadAsync()
+    {
+        // This is a defensive null check. It shouldn't be necessary, but prevents us from hooking up to the same data flow
+        // multiple times and, more importantly, losing track of links that such that we can no longer break them.
+        if (_launchSettingsLink is null)
+        {
+            // Here we use the DataflowUtilities.LinkToAsyncAction extension method to connect the data flow to the
+            // OnLaunchSettingsChangesAsync method, specifying the UnconfiguredProject as the "context" for the purposes of
+            // error reporting.
+            _launchSettingsLink = _launchSettingsProvider.SourceBlock
+                .LinkToAsyncAction(OnLaunchSettingsChangedAsync, _project);
```

Comment:
`LinkToAsyncAction` is defined in https://github.com/dotnet/project-system/blob/main/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DataflowUtilities.cs.

This in turn depends on [DataflowBlockFactory.CreateActionBlock](https://github.com/dotnet/project-system/blob/90cd51ccf5e7dd60f64f3e0d7b298c889eed80bc/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DataflowBlockFactory.cs#L17).